src/biogeochem/Fan2CTSMMod.F90

```diff
@@ -0,0 +1,1006 @@
+          ! Crop column, only one patch
+          p = col%patchi(c)
+          if (p /= col%patchf(c)) call endrun(msg='Strange patch for crop')
+          ratm = ram1(p) + rb1(p)
+       else
+          ! if natural, find average over grasses
+          ratm = 0.0
+          patchcounter = 0
+          do p = col%patchi(c), col%patchf(c)
+             if (patch%itype(p) == nc4_grass .or. patch%itype(p) == nc3_nonarctic_grass) then
+                if (.not. patch%active(p) .or. ram1(p) == spval .or. rb1(p) == spval) cycle
+                ratm = ratm + ram1(p) + rb1(p)
+                patchcounter = patchcounter + 1
+             end if
+          end do
+          if (patchcounter > 0) then
+             ratm = ratm / patchcounter
+          else
+             ! grass not found, take something.
+             do p = col%patchi(c), col%patchf(c)
+                if (.not. patch%active(p) .or. ram1(p) == spval .or. rb1(p) == spval) cycle
+                ratm = ram1(p) + rb1(p)
+                exit
+             end do
+             if (p == col%patchf(c) + 1) then
+                call endrun(msg='Could not find any useful pft for ram1')
+                ! Nothing found. We shouldn't be here.
+                ratm = 150.0_r8
+             end if
+          end if
+          ns%fan_grz_fract_col(c) = 1.0_r8 ! for crops handled by handle_storage
+       end if
+
+       watertend = waterstatebulk_inst%h2osoi_tend_tsl_col(c) * 1e-3 ! to meters/sec (ie. m3/m2/s)
+       
+       tg = temperature_inst%t_grnd_col(c)
+       theta = waterstatebulk_inst%h2osoi_vol_col(c,1)
+       thetasat = soilstate_inst%watsat_col(c,1)
+       bsw = soilstate_inst%bsw_col(c,1)
+       theta = min(theta, 0.98_r8*thetasat)
+       infiltr_m_s = max(waterfluxbulk_inst%qflx_infl_col(c), 0.0) * 1e-3 
+       evap_m_s = waterfluxbulk_inst%qflx_evap_grnd_col(c) * 1e-3
+       runoff_m_s = max(waterfluxbulk_inst%qflx_surf_col(c), 0.0) * 1e-3
+       if (runoff_m_s > 1.0_r8) then
+          runoff_m_s = 0.0_r8
+       end if
+       soilpsi = soilstate_inst%soilpsi_col(c,1)
+
+       ! grazing
+       !
+
+       ndep_org(ind_avail) = ngrz(c) * (1.0_r8-fract_tan) * fract_avail
+       ndep_org(ind_resist) = ngrz(c) * (1.0_r8-fract_tan) * fract_resist
+       ndep_org(ind_unavail) = ngrz(c) * (1.0_r8-fract_tan) * fract_unavail
+       tandep = ngrz(c) * fract_tan
+
+       orgpools(ind_avail) = man_a_grz(c)
+       orgpools(ind_resist) = man_r_grz(c)
+       orgpools(ind_unavail) = man_u_grz(c)
+       call update_org_n(ndep_org, tg, soilpsi, orgpools, dt, tanprod, soilflux_org, size(orgpools), status)
+       man_a_grz(c) = orgpools(ind_avail)
+       man_r_grz(c) = orgpools(ind_resist) 
+       man_u_grz(c) = orgpools(ind_unavail)
+
+       tanpools(1) = ns%tan_g1_col(c)
+       tanpools(2) = ns%tan_g2_col(c)
+       tanpools(3) = ns%tan_g3_col(c)
+
+       ph_soil = atm2lnd_inst%forc_soilph_grc(g)
+       if (ph_soil < 3.0) then
+          ph_soil = 6.5_r8
```

Comment:
There actually no pHs below 3 in the dataset. There was a problem that interpolating the pH from stream file gives invalid values for coastals gridpoints, because the values are missing for sea points. Since then, I switched to interpolating the streamfile outside CLM, which should avoid this problem. 

We should find a better way to do this. Basically, the interpolation should only use land points. Unless it is possible to somehow make the stream handle this, then one way is to extrapolate the maps over oceans from the nearest coastal point. I did this for the fertilizer types, and it looks stupid but works. 
![fract_urea in fan_nitrogen_f19_c20190325](https://user-images.githubusercontent.com/31929342/62394606-b3750780-b53b-11e9-891f-71b776a5896f.png)